src/libcore/task/wake.rs

```diff
@@ -119,7 +119,7 @@ impl LocalWaker {
     /// For this function to be used safely, it must be sound to call `inner.wake_local()`
     /// on the current thread.
     #[inline]
-    pub unsafe fn new(inner: NonNull<dyn UnsafeWake>) -> Self {
+    pub const unsafe fn new(inner: NonNull<dyn UnsafeWake>) -> Self {
```

Comment:
Please add the `rustc_const_unstable` attribute and a test enabling the feature and using it.

Reply:
See https://github.com/rust-lang/rust/pull/47562 for an example.

Reply:
Yep, I managed to get an example going. Unfortunately writing that example made me realise that this isn't usable for what I want anyway since it's `!Send + !Sync` so can't go in a static, as I don't know of any other usecases for this to be `const` I'm gonna close this and try to find an alternative.

One thing I noticed while adding the example, `rustc_const_unstable` requires a `stable`/`unstable` directly on the item being annotated, `LocalWaker` is in an `unstable` module so doesn't have a direct annotation, is there some reason the attribute needs to be directly on the item instead of inheriting from its parent scope?

Reply:
> Is there some reason the attribute needs to be directly on the item instead of inheriting from its parent scope?

Does implementation lazyness count as a reason?

Most newly added const fns already had a stability attribute, so it was less hazzle to add an attribute to those that didn't have it than to correctly implement the attribute

Reply:
Implementation lazyness is the best reason.

Reply:
*\<Joke about Haskell has been elided due to laziness\>*